src/DogStatsd.php

```diff
@@ -555,4 +568,17 @@ private function eventUdp($vals)
 
         return null;
     }
+
+    /**
+     * Normalize the value witout locale consideration before queuing the metric for sending
+     *
+     * @param string $value The value to normalize
+     *
+     * @return string Formatted value
+     */
+    protected function normalizeStat($value) {
+      // Controlls the way things are converted to a string.
+      // Otherwise localization settings impact float to string conversion (e.x 1.3 -> 1,3 and 10000 => 10,000)
+      return number_format($value, $this->decimalPrecision, '.', '');
```

Comment:
This force `.00` on all numbers and will increase the network usage for all payload for nothing. What do you think about trimming them ?

```php
return trim(number_format($value, $this->decimalPrecision, '.', ''), ".0");
```

Reply:
Needed a bit more logic than your snippet to trim, but should be all set now.

```
<?php

function normalizeStat($value) {
  // Controlls the way things are converted to a string.
  // Otherwise localization settings impact float to string conversion (e.x 1.3 -> 1,3 and 10000 => 10,000)
  $string_representation = rtrim(number_format($value, 5, '.', ''), "0");
  $last_char = substr($string_representation, -1, 1);
  if ($last_char == '.') {
    $string_representation = substr($string_representation, 0, strlen($string_representation) - 1);
  }
  return $string_representation;
}


echo normalizeStat(2.00) . PHP_EOL;
echo normalizeStat("96.000000") . PHP_EOL;
echo normalizeStat("2.006"). PHP_EOL;
echo normalizeStat(2.2222222222). PHP_EOL;
echo normalizeStat(2.01). PHP_EOL;
[rmadden@webphp-ftz2fns php-datadogstatsd]$ sudo php -f /tmp/normalizeTest.php
2
96
2.006
2.22222
2.01
```
Also tested for this in the tests i added, let me know if you need anything else.

Reply:
Yeah, trim's character class functionality has some edge case issues here since it chomps all consecutive characters of "0" or "." so 2000.00 becomes 2
```
TEST A
2
96
2.006
2.22222
2.01
2000
2000
TEST B
2
96
2.006
2.22222
2.01
2
2
[rmadden@webphp-ftz2fns php-datadogstatsd]$ cat /tmp/normalizeTest.php
<?php

function normalizeStatA($value) {
  // Controlls the way things are converted to a string.
  // Otherwise localization settings impact float to string conversion (e.x 1.3 -> 1,3 and 10000 => 10,000)
  $string_representation = rtrim(number_format($value, 5, '.', ''), "0");
  $last_char = substr($string_representation, -1, 1);
  if ($last_char == '.') {
    $string_representation = substr($string_representation, 0, strlen($string_representation) - 1);
  }
  return $string_representation;
}

function normalizeStatB($value) {
  // Controlls the way things are converted to a string.
  // Otherwise localization settings impact float to string conversion (e.x 1.3 -> 1,3 and 10000 => 10,000)
  return trim(number_format($value, 5, '.', ''), ".0");
}

echo "TEST A" . PHP_EOL;
echo normalizeStatA(2.00) . PHP_EOL;
echo normalizeStatA("96.000000") . PHP_EOL;
echo normalizeStatA("2.006"). PHP_EOL;
echo normalizeStatA(2.2222222222). PHP_EOL;
echo normalizeStatA(2.01). PHP_EOL;
echo normalizeStatA(2000). PHP_EOL;
echo normalizeStatA(2000.00). PHP_EOL;
echo "TEST B" . PHP_EOL;
echo normalizeStatB(2.00) . PHP_EOL;
echo normalizeStatB("96.000000") . PHP_EOL;
echo normalizeStatB("2.006"). PHP_EOL;
echo normalizeStatB(2.2222222222). PHP_EOL;
echo normalizeStatB(2.01). PHP_EOL;
echo normalizeStatB(2000). PHP_EOL;
echo normalizeStatB(2000.00). PHP_EOL;
```
Most concise way for me to show the difference between the functions